### DISCUSSION TITLE
Feature request: option to hide all UI

### DISCUSSION THREAD

**Discussion by liakhovitch** [flxzt/rnote#510] [Ideas]
The UI can get a little cramped on smaller devices, especially after 0.5.13. For example, the Surface Go line is meant to be the width of a vertical piece of 8.5x11 sheet when it is in _landscape_ (see attached image), and the various UI elements take away valuable canvas space. Some ideas on how to mitigate this:

- A "hide UI" button, which hides the entire UI (including titlebar). The UI can be returned via a single floating button, or perhaps some sort of gesture.
- An option to automatically hide the UI when a pen is active, and return it on a touch event. This would my personal favorite since I'm a pen-heavy user, but it may not work as well for people who do a lot of typing.

An idea on how _not_ to mitigate this:
- "Smart hide", where individual toolbars pop out from the side of the screen when the cursor is near them. This makes it tricky for touch users to make the bars show up, and for pen users it defeats the point of hiding the UI (ie, increasing usable canvas area)

![image](https://user-images.githubusercontent.com/22508850/218622960-828258de-df18-46a3-b231-b17e1cae6cf9.png)

**Comment by flxzt** [comment]
I like both suggestions, I wonder how to best implement this.

> A "hide UI" button, which hides the entire UI (including titlebar). The UI can be returned via a single floating button, or perhaps some sort of gesture.

I wonder if this should be combined with the fullscreen mode.

> An option to automatically hide the UI when a pen is active, and return it on a touch event. This would my personal favorite since I'm a pen-heavy user, but it may not work as well for people who do a lot of typing.

This is pretty specific to users with a 2in1 tablet (touch **and** stylus input). I am also thinking about users with an external drawing tablet and if there is a way to make it work for them as well. My first idea would be: hide the UI only while making progress with specific pens, e.g. hiding it while writing, drawing a shape, erasing and selecting, but showing it again as soon as those pens are "finished". There is the issue that the UI might appear under the pointer, so that is not ideal either.

---

What could also be added to improve the situation for smaller screens:

A different overlay layout for smaller devices (/ that can be enabled manually), that combines the toolbars into one at the top. This probably will need [AdwAdaptiveState](https://gitlab.gnome.org/GNOME/libadwaita/-/merge_requests/678)

Also: in the lastest beta (v0.5.14-beta1) the margins of the floating toolbars are reduced when scrollbars are disabled, so at least that gives a bit more canvas space.

Related is the discussion about the overlay toolbars in #367

**Reply by liakhovitch** [reply]
As far as 2-in-1 tablets go - wouldn't it work the same way? For a pen tablet user, the UI hides when they're using the pen and reappears when they use the mouse. Typing with the text tool might also trigger the UI hide. Of course, this doesn't work for people who use a pen as their main method of cursor input. For that reason, this behavior (if implemented) should probably be opt-in.

As for the compact overlay - if this becomes a thing, it would definitely help to be able to place it on either the top or the side. With a horizontal tablet, I would personally rather have the overlay on the side and get a bit of extra vertical space - but I'm sure some people would rather put it up top and get the longest line of handwriting possible. That's also why I'm suggesting that it may be simpler just to hide the UI entirely.

**Reply by flxzt** [reply]
This should definitely be optional (and easily accessible), but I was wondering what specific conditions would make sense. Agree with mouse & touch event, but maybe there can be more

But yeah regarding a compact UI, that is at least a few months away and there can definitely be both.

**Comment by flxzt** [comment]
continuing in #634
